core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt

```
@@ -0,0 +1,429 @@
+
+        val requiredFieldMessage = systemImpl.getString(MR.strings.field_required_prompt)
+
+        _uiState.update { prev ->
+            prev.copy(
+                firstNameError = if (savePerson.firstNames.isNullOrEmpty()) requiredFieldMessage else null,
+                genderError = if (savePerson.gender == GENDER_UNSET) requiredFieldMessage else null,
+                usernameError = if (!_uiState.value.signupWithPasskey && !validateUsername(
+                        savePerson.username ?: ""
+                    )
+                ) {
+                    systemImpl.getString(MR.strings.invalid)
+                } else {
+                    null
+                },
+                passwordError = if (!_uiState.value.signupWithPasskey && savePerson.username.isNullOrBlank()) {
+                    systemImpl.getString(MR.strings.field_required_prompt)
+                } else {
+                    null
+                }
+            )
+        }
+
+        if (_uiState.value.hasErrors()) {
+            loadingState = LoadingUiState.NOT_LOADING
+            return
+        }
+
+        viewModelScope.launch {
+
+
+            try {
+                val uid = activeDb.doorPrimaryKeyManager.nextIdAsync(Person.TABLE_ID)
+                savePerson.personUid = uid
+
+                if (_uiState.value.signupWithPasskey) {
+                    val passkeyCreated = createPasskeyUseCase?.invoke(
+                        CreatePasskeyParams
+                            (
+                            username = savePerson.firstNames.toString(),
+                            personUid = uid.toString(),
+                            doorNodeId = di.doorIdentityHashCode.toString(),
+                            usStartTime = systemTimeInMillis(),
+                            serverUrl = serverUrl,
+                            person = savePerson
+                        )
+                    )
+                    passkeyCreated?.let {
```

This will silently fail with no logging at all if the passkeycreated is null. We don't want that. we should show the user something went wrong and log it.